examples/mongodbatlas_federated_settings_identity_provider/azure/README.md

```diff
@@ -0,0 +1,233 @@
+# MongoDB Atlas Provider -- OIDC Workload Azure
+
+## Dependencies
+
+- Terraform [MongoDB Atlas Provider v1.17.0](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs)
+- Terraform [Hashicorp Cloud Init Provider v2.3.4](https://registry.terraform.io/providers/hashicorp/cloudinit/latest/docs)
+- Terraform [Hashicorp Azurerm Provider v3.106.1](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs)
+- Terraform [Hashicorp Random Provider v3.6.2](https://registry.terraform.io/providers/hashicorp/random/latest/docs)
+- A MongoDB Atlas account with an Organization configured with [Federation Authentication](https://www.mongodb.com/docs/atlas/security/federated-authentication/#federation-management-console)
+  - Get the `federated_settings_id` from the url, e.g., <https://cloud.mongodb.com/v2#/federation/{federated_settings_id}/overview>
+
+## Diagrams
+
+### Simplified OIDC flow
+
+
+```mermaid
+---
+title: "Simplified OIDC flow"
+---
+sequenceDiagram
+    autonumber
+    participant U as User/App
+    participant A as Atlas
+    participant Idp as Identity Provider (IdP)
+
+    Note right of A: OIDC is configured and set on the organization<br>issuer_uri=https://{some-idp}.com/
+    A->>Idp: issuer_uri/.well-known/openid-configuration
+    Idp->>A: {..., "jwks_uri": "jwks_uri"}
+    A->>Idp: jwks_uri
+    Idp->>A: {keys: [{"kid": kid}, ...]}
+
+    Note right of U: Obtain token
+    U->>Idp: get_token(authentication-mechanism)
+    Idp->>U: JWT token {"iss": "issuer_uri", sub: "user-identity", "kid": "key-used, ...}
+
+    Note right of A: A database user is registered with idp_id/sub as username
+    U->>A: connect(JWT token)
+    A->>A: use kid and keys to validate the token<br>lookup sub with registered users
+    A->>U: auth ok
+```
+
+- Example of `issuer_uri` (1):
+  - <https://token.actions.githubusercontent.com/>
+  - <https://gitlab.com/>
+  - <https://sts.windows.net/{tenant-id}/>
+- Example of authentication mechanism (5)
+  - username+password
+  - two factor authentication
+  - certificate
+- JWT Token Authentication(8):
+  - Can use different fields in the `JWT` to authenticate
+
+### OIDC Workload Azure
```

Comment:
[q] how different is Workforce example?

Reply:
I have been reading the [docs](https://www.mongodb.com/docs/atlas/workforce-oidc/#std-label-oidc-authentication-workforce); however, I'm not sure I understand it.
I believe:
- Workforce is for *humans*, and by enabling it, customers can log in to Atlas with their IdentityProvider. 
- However, could also be for connecting to the database directly.

@oarbusi do you know more?